File: SDKs/amazonaws.com/email/2010-12-01/java/lib/src/main/java/openapisdk/models/shared/TemplateMetadata.java
package openapisdk.models.shared;

import java.time.OffsetDateTime;

/**
 * TemplateMetadata
 * Contains information about an email template.
**/
public class TemplateMetadata {
    public OffsetDateTime createdTimestamp;
    public TemplateMetadata withCreatedTimestamp(OffsetDateTime createdTimestamp) {
        this.createdTimestamp = createdTimestamp;
        return this;
    }
    public String name;
    public TemplateMetadata withName(String name) {
        this.name = name;
        return this;
    }
}

File: SDKs/amazonaws.com/email/2010-12-01/python/src/sdk/models/shared/templatemetadata.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional


@dataclasses.dataclass
class TemplateMetadata:
    r"""TemplateMetadata
    Contains information about an email template.
    """
    
    created_timestamp: Optional[datetime] = dataclasses.field(default=None)
    name: Optional[str] = dataclasses.field(default=None)
    
